pkg/netutil/netutil.go

```diff
@@ -201,6 +204,18 @@ func urlsToStrings(us []url.URL) []string {
 	return rs
 }
 
+func stringsToURLs(us []string) ([]url.URL, error) {
+	urls := make([]url.URL, 0, len(us))
+	for _, str := range us {
+		u, err := url.Parse(str)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse string to URL: %q", str)
```

Comment:
It would be better to get the err included as well. 

```suggestion
			return nil, fmt.Errorf("failed to parse string to URL: %q, %w", str, err)
```

Since you are just backporting the PR, so please feel free to resolve this in a separate PR.

Reply:
I will create separate PR to fix this.